### PR TITLE
Update changelog for v2.4.0

### PR DESCRIPTION
**Pull request by demartinofra**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,38 @@ aws-parallelcluster-node CHANGELOG
 
 This file is used to list changes made in each version of the aws-parallelcluster-node package.
 
+2.4.0
+-----
+
+**ENHANCEMENTS**
+- dynamically fetch compute instance type and cluster size in order to support updates
```
Comment (enrico-usai) [inline CHANGELOG.md]: Capitalize: Dynamically

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,38 @@ aws-parallelcluster-node CHANGELOG
...
+  - scale up only if required slots/nodes can be satisfied
+  - scale down if pending jobs have unsatisfiable CPU/nodes requirements
+  - automatically terminate and replace faulty or unresponsive compute nodes
+
+**CHANGES**
+- SQS messages that fail to be processed are re-queued only 3 times and not forever
+- reset idletime to 0 when the host becomes essential for the cluster (because of min size of ASG or because there are
+  pending jobs in scheduler queue)
+- SGE: a node is considered as busy when in one of the following states "u", "C", "s", "d", "D", "E", "P", "o".
+  This allows an quick replacement of the node without waiting for the nodewatcher to terminate it.
```
Comment (enrico-usai) [inline CHANGELOG.md]: typo: an --> a

File: CHANGELOG.md (hunk 1 above)
Comment (demartinofra) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 2 above)
Comment (demartinofra) [reply CHANGELOG.md]: fixed

**Pull request merged**